spec/queue-spec.js

```diff
@@ -0,0 +1,68 @@
+import Q from 'q'
+import {expect} from 'chai'
+import Queue from '../src/queue.js'
+import Task from '../src/task.js'
+
+class SampleTask extends Task {
+
+  constructor(time) {
+    super()
+    this.time = time
+  }
+
+  run() {
+    const deferred = Q.defer()
+    setTimeout(() => deferred.resolve(), this.time)
+    return deferred.promise
+  }
+
+  done() {
+    // Job done!
+  }
+
+}
+
+describe('Queue', () => {
+
+  it('should add a job to the queue', (done) => {
+    expect(Queue.tasks.length).to.equal(0)
+    Queue.push(new SampleTask(100))
+    expect(Queue.tasks.length).to.equal(1)
+    setTimeout(done, 200)
+  })
+
+  it('should run a task added to the queue', (done) => {
+    const time = 200;
+    expect(Queue.running).to.be.false
+    Queue.push(new SampleTask(time))
+    expect(Queue.running).to.be.true
+    setTimeout(() => {
+      expect(Queue.tasks.length).to.equal(0)
+      expect(Queue.running).to.be.false
+      done()
+    }, time + 100)
+  })
+
+  it('should add non-specified tasks to "generic" queue', () => {
+    Queue.push(new SampleTask(100))
+    expect(Queue.pendingCounter.generic).to.not.be.undefined
+    expect(Queue.pendingCounter.generic).to.equal(1)
+    expect(Queue.running).to.be.true
+    setTimeout(() => {
+      expect(Queue.pendingCounter.generic).to.equal(0)
+      expect(Queue.running).to.be.false
+    }, 150)
+  })
+
+  it('should add specific tasks to specific queue', () => {
+    Queue.pushTo('sample', new SampleTask(100))
+    expect(Queue.pendingCounter.sample).to.not.be.undefined
+    expect(Queue.pendingCounter.sample).to.equal(1)
+    expect(Queue.running).to.be.true
+    setTimeout(() => {
+      expect(Queue.pendingCounter.sample).to.equal(0)
+      expect(Queue.running).to.be.false
+    }, 150)
```

Comment:
What about creating a local constant to save those timing numbers, avoiding magic numbers around the test suite? May be a good low hanging fruit.